### PR TITLE
Use Terraform SDK v2

### PR DESCRIPTION
**Pull request by vincentbernat**
This is really a matter of using sed to replace the imports and a tiny
change in `main.go`. I need this to build a Pulumi provider with the
Terraform bridge.

### CODE REVIEW

**Comment by vincentbernat** [comment]
Note that I have not tested with Terraform directly. It works with Pulumi, but I didn't do extensive tests. Please test!

**Review by nlewo — APPROVED** [review]
Built and tested on the simplehosting API.
LGTM.

And thank you @vincentbernat !

File: gandi/data_domain.go
Comment (taoufik07) [inline gandi/data_domain.go:3]: @tiramiseb @nlewo I think we should first extensively test (sadly manually) and release (or at least release) the current master, before merging this PR

**Review by taoufik07 — CHANGES_REQUESTED** [review]

File: gandi/data_domain.go
Comment (taoufik07) [reply gandi/data_domain.go:3]: (The rc is just to block the merge, if you disagree, let me know and I'll dismiss my request change)

Comment (tiramiseb) [reply gandi/data_domain.go:3]: Is there any reason the current master must be releases ? Couldn't the first release be with the SDK v2?

**Comment by vincentbernat** [comment]
I have pushed the easy part of the changes about deprecated functions. The retry part is more mysterious as I don't have a context to propagate and just putting `context.Background()` or `context.TODO()` does not seem great.

**Comment by vincentbernat** [comment]
I have added another commit for replacement of `resource.Retry()` as it requires more changes. I can squash if needed.

**Comment by tiramiseb** [comment]
I'll rely on you @nlewo for this, I didn't take a look on what would be impacted etc.

@taoufik07 could you please explain why you would prefer a release before this PR? thanks

File: gandi/data_domain.go
Comment (taoufik07) [reply gandi/data_domain.go:3]: Last time I've checked, the v2 shiped with some breaking changes, not sure if we're concerned, ofc we can read the changelog and verify if we're concerned, but since we don't have tests, no one can guarantee that it works, I think it's safer to first draft a release without these changes.

And we really need a release, just a simple tag will do for now, since we make install manually, because every time a new commit is pushed, code that depends on it is likely to break, I personally checkout to a commit before make install, which's not sustainable, since I have to keep track of the commit across various CI jobs and test all our infra every time I try to update the commit and it's also hard for new comers to tell that the safer is to find a commit that works and pin it.

I can volunteer to draft a release if you don't have time.

Comment (nlewo) [reply gandi/data_domain.go:3]: I'm actually pretty confident with this change.

The [upgrade guide](https://www.terraform.io/plugin/sdkv2/guides/v2-upgrade-guide) doesn't contain any warning about this upgrade. There is a list of breaking changes in the [release notes](https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.0.0/CHANGELOG.md) and i don't see anything that could impact us (other important providers no longer support terraform 0.11.).

@taoufik07 But as you suggested, i'll do more manual tests of this branch.

Comment (taoufik07) [reply gandi/data_domain.go:3]: Well then 🤞

Comment (taoufik07) [reply gandi/data_domain.go:3]: Well then 🤞

**Comment by nlewo** [comment]
I tried to manage Simple Hosting (with retries), Domain, LiveDNS records. Everything worked as expected.

So, i propose to merge it by tomorrow if nobody complains.

**Comment by tiramiseb** [comment]
I approve waiting until tomorrow wed. 29th.

@nlewo, don't hesitate to merge tomorrow, I'm okay

**Comment by robbyoconnor** [comment]
@nlewo did you try importing an existing `gandi_livedns_domain`? That would fix #57 if that worked.

**Comment by nlewo** [comment]
@robbyoconnor No, i didn't try but i'm pretty sure it won't fix it (cf [my comment](https://github.com/go-gandi/terraform-provider-gandi/issues/55#issuecomment-1002937755) ).

**Pull request merged**
